File: Java/7-Tutorial/2/NegativeExpression.java
// Nama		: Naufal Dean Anugrah
// NIM		: 13518123
// Tanggal	: 12 Maret 2020
// Topik 	: Pengenalan java

class NegativeExpression implements Expression {
	protected Expression x;

	public NegativeExpression(Expression x) {
		this.x = x;
	}

	public int solve() {
		return -1 * this.x.solve();
	}
}

File: C++/4-Praktikum/4/NegativeExpression.cpp
// Nama         : Naufal Dean Anugrah
// NIM          : 13518123
// Tanggal      : 13 Februari 2020 
// Topik        : Inheritance, Polymorphism, Abstract Base Class

#include "NegativeExpression.hpp"


NegativeExpression::NegativeExpression(Expression * T1) : UnaryExpression(T1) {}

int NegativeExpression::solve() {
	return (-1) * x->solve();
}
